geode-cq/src/main/java/org/apache/geode/cache/query/cq/internal/ServerCQImpl.java

```diff
@@ -203,7 +203,7 @@ public void registerCq(ClientProxyMembershipID p_clientProxyId, CacheClientNotif
     this.updateCqCreateStats();
 
     // Initialize the state of CQ.
-    if (this.cqState.getState() != p_cqState) {
+    if (this.cqState.getState() != p_cqState || ccn == null) {
```

Comment:
Please rename `ccn` to `cacheClientNotifier`... Just small changes to increase the code cleanliness.